src/v5_hal/firmware/src/nodes/subsystems/drivetrain_nodes/HolonomicDriveNode.cpp

```diff
@@ -0,0 +1,103 @@
+    m_left_front_motor->resetEncoder();
+    m_left_rear_motor->resetEncoder();
+    m_right_front_motor->resetEncoder();
+    m_right_rear_motor->resetEncoder();
+}
+
+void HolonomicDriveNode::m_setLeftFrontVoltage(int voltage) {
+    m_left_front_motor->moveVoltage(voltage);
+}
+
+void HolonomicDriveNode::m_setLeftRearVoltage(int voltage) {
+    m_left_rear_motor->moveVoltage(voltage);
+}
+
+void HolonomicDriveNode::m_setRightFrontVoltage(int voltage) {
+    m_right_front_motor->moveVoltage(voltage);
+}
+
+void HolonomicDriveNode::m_setRightRearVoltage(int voltage) {
+    m_right_rear_motor->moveVoltage(voltage);
+}
+
+void HolonomicDriveNode::m_setLeftFrontVelocity(int velocity) {
+    m_left_front_motor->moveVelocity(velocity);
+}
+
+void HolonomicDriveNode::m_setLeftRearVelocity(int velocity) {
+    m_left_rear_motor->moveVelocity(velocity);
+}
+
+void HolonomicDriveNode::m_setRightFrontVelocity(int velocity) {
+    m_right_front_motor->moveVelocity(velocity);
+}
+
+void HolonomicDriveNode::m_setRightRearVelocity(int velocity) {
+    m_right_rear_motor->moveVelocity(velocity);
+}
+
+int HolonomicDriveNode::getLeftFrontPosition() {
+    return m_left_front_motor->getPosition();
+}
+
+int HolonomicDriveNode::getLeftRearPosition() {
+    return m_left_rear_motor->getPosition();
+}
+
+int HolonomicDriveNode::getRightFrontPosition() {
+    return m_right_front_motor->getPosition();
+}
+
+int HolonomicDriveNode::getRightRearPosition() {
+    return m_right_rear_motor->getPosition();
+}
+
+void HolonomicDriveNode::initialize() {
+    resetEncoders();
+}
+
+void HolonomicDriveNode::setDriveVoltage(int left_front_voltage, int left_rear_voltage, 
+    int right_front_voltage, int right_rear_voltage) {
+    m_setLeftFrontVoltage(left_front_voltage);
+    m_setLeftRearVoltage(left_rear_voltage);
+    m_setRightFrontVoltage(right_front_voltage);
+    m_setRightRearVoltage(right_rear_voltage);    
+}
+
+void HolonomicDriveNode::setDriveVelocity(float left_front_velocity, float left_rear_velocity, 
+    float right_front_velocity, float right_rear_velocity) { //incoming values should be in m/s so we convert to rpm here
+    m_setLeftFrontVelocity(left_front_velocity / MAX_WHEEL_SPEED * 200);
+    m_setLeftRearVelocity(left_rear_velocity / MAX_WHEEL_SPEED * 200);
+    m_setRightFrontVelocity(right_front_velocity / MAX_WHEEL_SPEED * 200);
+    m_setRightRearVelocity(right_rear_velocity / MAX_WHEEL_SPEED * 200);
+}
+
+void HolonomicDriveNode::teleopPeriodic() {
+    m_setLeftFrontVoltage(((m_controller->get_analog(pros::E_CONTROLLER_ANALOG_LEFT_Y)) / 127.0) * MAX_MOTOR_VOLTAGE);
+    m_setLeftRearVoltage(((m_controller->get_analog(pros::E_CONTROLLER_ANALOG_LEFT_Y)) / 127.0) * MAX_MOTOR_VOLTAGE);
+    m_setRightFrontVoltage(((m_controller->get_analog(pros::E_CONTROLLER_ANALOG_LEFT_X)) / 127.0) * MAX_MOTOR_VOLTAGE);
+    m_setRightRearVoltage(((m_controller->get_analog(pros::E_CONTROLLER_ANALOG_LEFT_X)) / 127.0) * MAX_MOTOR_VOLTAGE);
```

Comment:
The code for controlling the holonomic drive will be a little bit different. This outline from VEX Coding studio will need to be changed around slightly but should have the basics correct: 
![image](https://user-images.githubusercontent.com/6099416/111051439-9b07e700-8418-11eb-99ce-72acf919a612.png) - Let us know if you have any questions about the implementation or the channels being used

Reply:
The channels are on the controller in red (by the joysticks): 
![image](https://user-images.githubusercontent.com/6099416/111051457-c7236800-8418-11eb-80bd-23a5e32b83fe.png)